swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPublic.qll

```diff
@@ -51,6 +51,11 @@ class Node extends TNode {
    * Gets this node's underlying SSA definition, if any.
    */
   Ssa::Definition asDefinition() { none() }
+
+  /**
+   * Gets the parameter that corresponds to this node, if any.
+   */
+  ParamDecl getParameter() { none() }
```

Comment:
Shouldn't this predicate be called `asParameter` for consistency? I.e. to indicate that it corresponds to a cast more than a getter.
```suggestion
  ParamDecl asParameter() { result = this.(ParameterNode).getParameter() }
```

Reply:
Indeed, it should. I'll fix that up in a commit since we need to rename a use of it as well.

Reply:
Turns out there was a lot of changes necessary here. There seems to have been some confusion about `asParameter` (for getting the node corresponding to a parameter) and `getParameter` (for any kind of dataflow node that happened to be defined in terms of a parameter). This has been fixed in 951b6beeb150f5b0db1e20a50ce52a2094d76894. Thanks for bringing this to my attention!